core/src/com/mygdx/crazybomber/model/player/Player.java

```diff
@@ -137,20 +140,18 @@ public Player(double playerSpawnXCoordinate, double playerSpawnYCoordinate, Map
         getMap().getActiveBombArray().add(bomb);
     }
 
-    public void addBomb(Map map) {
-        final Bomb newBomb = new Bomb(this, _bombStack);
-        this._bombStack.push(newBomb);
-    }
-
-    public void pickUpSpeedUpgrade() {
-        setSpeed(getSpeed() + 1.0);
-    }
-
-    public void pickUpRangeUpgrade() {
-        setNumRangeUpgrades(getNumRangeUpgrades() + 1);
+    public void pickUpItem(Item item, Map map) {
```

Comment:
this needs to be edited such that it sends data to the server -> accepts if the server approves the item pickup or not -> then do the logic shown here.